test/lib/code_corps_web/controllers/user_controller_test.exs

```diff
@@ -243,26 +245,33 @@ defmodule CodeCorpsWeb.UserControllerTest do
     end
   end
 
-
   describe "github_oauth" do
-    test "return the user when current user connects successfully", %{conn: conn} do
-      user = insert(:user)
+    @tag :authenticated
+    test "return the user when current user connects successfully", %{conn: conn, current_user: current_user} do
+      attrs = %{"code" => "valid_code", "state" => "valid_state"}
 
-      json = %{"code" => "valid_code", "state" => "valid_state"}
+      path = user_path(conn, :github_oauth)
 
-      path = user_path(conn, :github_oauth, json)
+      json = conn |> post(path, attrs) |> json_response(200)
 
-      json = conn |> authenticate(user) |> post(path) |> json_response(200)
+      assert json["data"]["id"] |> String.to_integer == current_user.id
+      assert json["data"]["attributes"]["github-id"]
+    end
 
-      assert json["data"]["id"] |> String.to_integer == user.id
+    test "requires authentication", %{conn: conn} do
+      path = user_path(conn, :github_oauth, %{"code" => "foo", "state" => "bar"})
```

Comment:
Unless I'm misreading this entirely, I think what's confusing me about this test is that it implies that `valid_code` and `valid_state` somehow work whereas `foo` and `bar` would not, when in reality it simply tests that the user is authenticated with our system regardless of the params that make it to GitHub.

Can you clarify @begedin?

Reply:
It should work regardless, I'll update.

Reply:
@JoshSmith To clarify, the controller action was implemented with the `code` and `state` in the signature. I've moved `attrs` into a module variable, so it doesn't imply that specific values are important. 

However, the action will only work with attributes containing `code` and `state` keys.

Not sure if I should loosen that requirement, or write a test for it. Either way, this PR is about closing #790. If we want to loosen the requirement, then that requires a rewrite of `GitHub.User` as well, so I would do it separately. Adding a test could go here, but this would be the first time we're writing a test for such a case.

Reply:
This addresses totally what I was taking issue with, thanks!